### ISSUE TITLE
r10k doesn't properly handle changes to a module's remote

### ISSUE THREAD

**Issue by ghost** [puppetlabs/r10k#131]
Recently, I had a situation come up where I had to change a git remote and ref on a module specified in our Puppetfile. Upon changing the `jenkins` module from `git.github.com/sschneid/puppet-jenkins.git` to `git.github.com/jenkinsci/puppet-jenkins.git`, and updating `:ref`, this happened:

```
[R10K::Task::Module::Sync - INFO] Deploying jenkins into /etc/puppet/environments/puppetize_jenkins/modules
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x7f9a52b14c18> failed while running: Cannot check out unresolvable ref 'v1.0.1' at /etc/puppet/environments/puppetize_jenkins/modules/jenkins
```

Further digging into the issue, it looks like r10k realized that the module already existed and tried checking out a non-existent ref:

```
root@ningyo:/etc/puppet/environments/puppetize_jenkins/modules/jenkins# git remote -v show
cache   /var/cache/r10k/git---github.com-sschneid-puppet-jenkins.git (fetch)
cache   /var/cache/r10k/git---github.com-sschneid-puppet-jenkins.git (push)
origin  git://github.com/sschneid/puppet-jenkins.git (fetch)
origin  git://github.com/sschneid/puppet-jenkins.git (push)
```

It might be worthwhile validating that the remotes are still sane, and if they are, check out the ref. If they aren't, blow away the module and re-clone it as if it never existed.

This one is pretty easy for me to reproduce, so let me know if you'd like more info.

**Comment by adrienthebo** [comment]
@rji I merged the fix for GH-137 into 1.3.x, could you test that branch and verify that this is fixed?

**Comment by adrienthebo** [comment]
I spoke with @acidprime and he's pretty confident that the fix for GH-137 should cover this well. I'm going to close this out with the assumption it's fixed and release 1.2.2; if this hasn't been fixed after all let me know and I can reopen the ticket.
